my-app/src/components/Login.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import "./Login.css";
+import axiosWithAuth from "./utils/AxiosWithAuth";
+import { gsap } from "gsap";
+import { Link } from "react-router-dom";
+import * as yup from "yup";
+
+const initialState = {
+  username: "",
+  password: "",
+  isFetching: false,
+};
+
+const initialFormErrors = {
+  username: "Username is required!",
+  password: "Password is required!",
+};
+
+const loginSchema = yup.object().shape({
+  username: yup
+    .string()
+    .min(3, "Username must include at least 3 characters!")
+    .required("Username is required!"),
+  password: yup
+    .string()
+    .min(5, "Password must include at least 5 characters!")
+    .required("Password is required!"),
+});
+
+export const Login = (props) => {
+  const [login, setLogin] = useState(initialState);
+  const [loginFormErrors, setLoginFormErrors] = useState(initialFormErrors);
+  const [buttonEnabled, setButtonEnabled] = useState(false);
+
+  useEffect(() => {
+    gsap.from(".login", { x: -500, duration: 1.0, ease: "expo.out" });
+    gsap.fromTo(".login", { autoAlpha: 0 }, { autoAlpha: 1, duration: 1.0 });
+  }, []);
+
+  useEffect(() => {
+    loginSchema.isValid(login).then((valid) => {
+      setButtonEnabled(valid);
+    });
+  }, [login]);
+
+  const handleChange = (e) => {
+    e.persist();
+    setLogin({ ...login, [e.target.name]: e.target.value });
+
+    yup
+      .reach(loginSchema, e.target.name)
+      .validate(e.target.value)
+      .then((valid) => {
+        setLoginFormErrors({ ...loginFormErrors, [e.target.name]: "" });
+      })
+      .catch((err) => {
+        setLoginFormErrors({
+          ...loginFormErrors,
+          [e.target.name]: err.errors[0],
+        });
+      });
+  };
+
+  const reloadPage = () => {
+    window.location.reload();
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    axiosWithAuth()
+      .post("/login", login)
+      // axios.post('https://secret-family-recipes-bw-team5.herokuapp.com/login', login)
+      .then((res) => {
+        props.setUserId(res.data.id);
+        console.log("Login data returning", res);
+        props.history.push("/recipe");
```

Comment:
Change this to:
```
props.history.push(`/${res.data.id}/recipe`);
```
Since state isn't updating fast enough we will route using the id returned in res.  We still want to set it in state to use later.